nectar/src/ethereum/wallet.rs

```diff
@@ -355,6 +321,36 @@ impl Wallet {
             .await
     }
 
+    async fn get_transaction_by_hash(&self, transaction_hash: Hash) -> anyhow::Result<Transaction> {
+        self.geth_client
+            .get_transaction_by_hash(transaction_hash)
+            .await
+    }
+
+    async fn wait_until_transaction_receipt(
+        &self,
+        transaction_hash: Hash,
+        chain: ChainId,
+    ) -> anyhow::Result<TransactionReceipt> {
+        let poll_interval = match chain {
+            ChainId::MAINNET => 10, // roughly half the blocktime
+            ChainId::KOVAN => 2,    // roughly half the blocktime
+            ChainId::ROPSTEN => 10, // kovan has inconsistent blocktime between 1 sec and 1 min
```

Comment:
```suggestion
            ChainId::ROPSTEN => 10, // ropsten has inconsistent blocktime between 1 sec and 1 min
```

Or did you actually mean Kovan?

Reply:
I did not :)